File: tests/ArrayIndexOf.cs
using System.Text;

namespace Demo {
    public partial class Utils {
        public void Sanitize(char ch, char[] invalidChars, StringBuilder sanitizedInput) {
            if (Array.IndexOf(invalidChars, ch) == -1) sanitizedInput.Append(ch); else sanitizedInput.Append("_");
        }
    }
}

File: tests/ArrayIndexOf.pas
namespace Demo;

type
  Utils = class
  public
    method Sanitize(ch: Char; invalidChars: array of Char; sanitizedInput: StringBuilder);
  end;

implementation

uses System.Text;

method Utils.Sanitize(ch: Char; invalidChars: array of Char; sanitizedInput: StringBuilder);
begin
  if Array.IndexOf(invalidChars, ch) = -1 then
    sanitizedInput.Append(ch)
  else
    sanitizedInput.Append('_');
end;

end.
